src/core_modules/capture-core/components/WidgetProfile/DataEntry/factory/DataElementFactory.js

```diff
@@ -0,0 +1,196 @@
+
+            o.onValidate = (value: any, contextProps: Object = {}) => {
+                const serverValue = pipe(convertFormToClient, convertClientToServer)(
+                    value,
+                    cachedTrackedEntityAttribute.valueType,
+                );
+
+                if (contextProps.onGetUnsavedAttributeValues) {
+                    const unsavedAttributeValues = contextProps.onGetUnsavedAttributeValues(dataElement.id);
+                    if (unsavedAttributeValues) {
+                        const foundValue = unsavedAttributeValues.find(usav => usav === serverValue);
+                        if (foundValue) {
+                            return {
+                                valid: false,
+                                data: {
+                                    attributeValueExistsUnsaved: true,
+                                },
+                            };
+                        }
+                    }
+                }
+
+                let requestPromise;
+                if (o.scope === dataElementUniqueScope.ORGANISATION_UNIT) {
+                    const orgUnitId = contextProps.orgUnitId;
+                    requestPromise = getApi().get('trackedEntityInstances', {
+                        program: contextProps.programId,
+                        ou: orgUnitId,
+                        filter: `${dataElement.id}:EQ:${serverValue}`,
+                    });
+                } else {
+                    requestPromise = getApi().get('trackedEntityInstances', {
+                        program: contextProps.programId,
+                        ouMode: 'ACCESSIBLE',
+                        filter: `${dataElement.id}:EQ:${serverValue}`,
+                    });
+                }
+                return requestPromise.then((result) => {
+                    const trackedEntityInstance =
+                        (result.trackedEntityInstances && result.trackedEntityInstances[0]) || {};
+                    const data = {
+                        id: trackedEntityInstance.trackedEntityInstance,
+                        tetId: trackedEntityInstance.trackedEntityType,
+                    };
+
+                    return {
+                        valid: result.trackedEntityInstances.length === 0,
+                        data,
+                    };
+                });
+            };
+
+            if (cachedTrackedEntityAttribute.pattern) {
+                o.generatable = !!cachedTrackedEntityAttribute.pattern;
+            }
+        });
+    }
+
+    optionSetFactory: OptionSetFactory;
+    cachedTrackedEntityAttributes: Map<string, CachedTrackedEntityAttribute>;
+    constructor({ cachedTrackedEntityAttributes, cachedOptionSets }: ConstructorInput) {
+        this.cachedTrackedEntityAttributes = cachedTrackedEntityAttributes;
+        this.optionSetFactory = new OptionSetFactory(cachedOptionSets);
+    }
+
+    // eslint-disable-next-line complexity
```

Comment:
why do we need to disable complexity here, is there any way to split the function to smaller ones?